test/extended/util/nodes.go

```diff
@@ -95,19 +95,30 @@ func DebugAllNodesRetryWithOptionsAndChroot(oc *CLI, debugNodeNamespace string,
 // TriggerNodeRebootGraceful initiates a graceful node reboot which allows the system to terminate processes cleanly before rebooting.
 func TriggerNodeRebootGraceful(kubeClient kubernetes.Interface, nodeName string) error {
 	command := "echo 'reboot in 1 minute'; exec chroot /host shutdown -r 1"
-	return triggerNodeReboot(kubeClient, nodeName, 0, command)
+	return triggerNodeDisruption(kubeClient, nodeName, 0, false, command)
 }
 
 // TriggerNodeRebootUngraceful initiates an ungraceful node reboot which does not allow the system to terminate processes cleanly before rebooting.
 func TriggerNodeRebootUngraceful(kubeClient kubernetes.Interface, nodeName string) error {
 	command := "echo 'reboot in 1 minute'; exec chroot /host sudo systemd-run sh -c 'sleep 60 && reboot --force --force'"
-	return triggerNodeReboot(kubeClient, nodeName, 0, command)
+	return triggerNodeDisruption(kubeClient, nodeName, 0, false, command)
 }
 
-func triggerNodeReboot(kubeClient kubernetes.Interface, nodeName string, attempt int, command string) error {
+// TriggerNetworkDisruption temporarily disables the specified network interface on the target node
+func TriggerNetworkDisruption(kubeClient kubernetes.Interface, target, peer corev1.Node, networkInterface string, disruptionDuration time.Duration) (string, error) {
+	command_preable := fmt.Sprintf("echo 'temporarily disabling interface %s on %s for %v'; exec chroot /host sh -c ", networkInterface, target.Name, disruptionDuration)
+	interface_down_cmd := fmt.Sprintf("ip link set %s down", networkInterface)
+	timeout_cmd := fmt.Sprintf("sleep %d", int(disruptionDuration.Seconds()))
+	interface_up_cmd := fmt.Sprintf("ip link set %s up", networkInterface)
+	command := fmt.Sprintf("%s '%s ; %s ; %s'", command_preable, interface_down_cmd, timeout_cmd, interface_up_cmd)
+
+	return command, triggerNodeDisruption(kubeClient, target.Name, 0, true, command)
+}
+
+func triggerNodeDisruption(kubeClient kubernetes.Interface, nodeName string, attempt int, useHostNetwork bool, command string) error {
```

Comment:
I don't love the `userHostNetwork` bool.
I would prefer some kind of enum that allows me to specify which kind of disruption.
-> NodeCrash
-> NetworkPartition
etc.

I think this will allow us to reuse the pattern as we add more disruption types (e.g. kubelet failure, etc)

Reply:
I get what you mean. Maybe a name tweak would suffice. Instead of "triggerNodeDisruption", could we rename the function to something like: "createNodeDisruptionPod" or "execNodeDisruptionPod"

This would make it clear to that usingHostNetwork is a change that affects the created pod, not something that dictates the kind of disruption. (The kind of disruption is defined by the command / all of the arguments together).

Reply:
Good idea :+1: